_rules/image-filename-as-accessible-name-9eb3f6.md

```diff
@@ -61,45 +68,88 @@ _There are no major accessibility support issues known for this rule._
+This `img` element has an [accessible name][] equivalent to the filename (ignoring letter casing). The [accessible name][] accurately describes the image.
 
 ```html
 <html lang="en">
-	<img src="https://www.w3.org/WAI/demos/bad/img/w3c" alt="w3c" />
+	<img src="/test-assets/image-filename-as-accessible-name-9eb3f6/nyhavn" alt="Nyhavn" />
 </html>
 ```
 
 #### Passed Example 2
 
-This `img` element has an [accessible name][] equivalent to the filename. The [accessible name][] in combination with the text content of the `a` element accurately describes the image.
+This `img` element has an [accessible name][] equivalent to the filename. Because the image is used in a download link, the presence of the file extension is a relevant part of its description.
 
 ```html
 <html lang="en">
-	<a href="https://www.w3.org/WAI/demos/bad/img/w3c.png" download
-		>Download <img src="https://www.w3.org/WAI/demos/bad/img/w3c.png" alt="w3c.png"
+	<a href="/test-assets/image-filename-as-accessible-name-9eb3f6/nyhavn.jpeg" download
+		>Download <img src="/test-assets/image-filename-as-accessible-name-9eb3f6/nyhavn.jpeg" alt="nyhavn.jpeg"
 	/></a>
 </html>
 ```
 
+#### Passed Example 3
+
+This image button has an [accessible name][] equivalent to the filename. The [accessible name][] accurately describes the purpose of the button.
+
+```html
+<html lang="en">
+	<input type="image" src="test-assets/image-filename-as-accessible-name-9eb3f6/login" alt="login" />
+</html>
+```
+
+#### Passed Example 4
+
+This `img` element has an [accessible name][] equivalent to the filename. The [accessible name][] accurately describes the image in the language of the element (French, same as the language of the page).
+
+```html
+<html lang="fr">
+	<img src="test-assets/image-filename-as-accessible-name-9eb3f6/pain" alt="pain" />
+</html>
+```
+
 ### Failed
 
 #### Failed Example 1
 
-This `img` element has [accessible name][] matching the image filename. The presence of the file extension in the [accessible name][] is redundant and results in the [accessible name][] not accurately describing the image.
+This `img` element has [accessible name][] matching the image filename (ignoring letter casing). The name does not describe the image.
 
 ```html
 <html lang="en">
-	<img src="https://www.w3.org/WAI/demos/bad/img/w3c.png" alt="w3c.png" />
+	<img src="/test-assets/image-filename-as-accessible-name-9eb3f6/paris" alt="Paris" />
 </html>
 ```
 
 #### Failed Example 2
 
-This `input` element with a `type` of `image` has a [semantic role][] of `img` and an [accessible name][] matching the filename. The presence of the file extension in the [accessible name][] is redundant and results in the [accessible name][] not accurately describing the image.
+This `img` element has [accessible name][] matching the image filename. The name is just a checksum and does not describe the image.
```

Comment:
What's the purpose of this example? What does it add that isn't already covered?

Reply:
I agree that it is similar to the previous one.
Such a case was mention in the TF review by @maryjom so I thought it might be good to have it.
It showcases a common case of very bad filename (while the first failed example is a bit unlikely but is better at showing that the rule requires a **descriptive** accessible name, not just a **meaningful** one).

Reply:
OK sure, why not I suppose.